From 29e135bea4156e9a677ba44148537f7375841020 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Mon, 13 Dec 2010 15:47:28 +0800
Subject: [PATCH] ENGR00136875-1 Add function pgprot_writethru()

Added pgprot_writethru(), to set the buffer's cache property as
writethrough.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 arch/arm/include/asm/pgtable.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Index: linux-2.6.38.8/arch/arm/include/asm/pgtable.h
===================================================================
--- linux-2.6.38.8.orig/arch/arm/include/asm/pgtable.h	2011-06-03 03:35:11.000000000 +0200
+++ linux-2.6.38.8/arch/arm/include/asm/pgtable.h	2012-03-13 18:20:29.000000000 +0100
@@ -232,6 +232,9 @@
 #define pgprot_writecombine(prot) \
 	__pgprot_modify(prot, L_PTE_MT_MASK, L_PTE_MT_BUFFERABLE)
 
+#define pgprot_writethru(prot) \
+	__pgprot_modify(prot, L_PTE_MT_MASK, L_PTE_MT_WRITETHROUGH)
+
 #ifdef CONFIG_ARM_DMA_MEM_BUFFERABLE
 #define pgprot_dmacoherent(prot) \
 	__pgprot_modify(prot, L_PTE_MT_MASK, L_PTE_MT_BUFFERABLE | L_PTE_XN)
